DOCSUP-5822: Document the accurateCastOrNull function. (#19965)
ClickHouse/ClickHouse: docs/en/sql-reference/functions/type-conversion-functions.md

Accepts a String or FixedString argument. Returns the String with the content truncated at the first zero byte found.

Example:
**Example**

Query:

``` sql
SELECT toFixedString('foo', 8) AS s, toStringCutToZero(s) AS s_cut
SELECT toFixedString('foo', 8) AS s, toStringCutToZero(s) AS s_cut;
```

Result:

``` text
┌─s─────────────┬─s_cut─┐
│ foo\0\0\0\0\0 │ foo   │
└───────────────┴───────┘
```

Query:

``` sql
SELECT toFixedString('foo\0bar', 8) AS s, toStringCutToZero(s) AS s_cut
SELECT toFixedString('foo\0bar', 8) AS s, toStringCutToZero(s) AS s_cut;
```

Result:

``` text
┌─s──────────┬─s_cut─┐
│ foo\0bar\0 │ foo   │
└────────────┴───────┘
```

> ## reinterpretAs(x, T) {#type_conversion_function-cast}

Reviewer: This variant still works for me. It is strange it was removed.

## CAST(x, T) {#type_conversion_function-cast}

Converts input value `x` to the `T` data type. The syntax `CAST(x AS t)` is also supported.

> 

Reviewer 2: Note, that if value `x` does not fit the bounds of type `T`, the function will overflow. For example, `CAST(-1, 'UInt8')` will return `255`.

## accurateCast(x, T) {#type_conversion_function-accurate-cast}

> Converts `x` to the `T` data type. The differente from [cast(x, T)](#type_conversion_function-cast) is that `accurateCast`
> does not allow overflow of numeric types during cast if type value `x` does not fit bounds of type `T`.

Reviewer 2: ```suggestion
Converts `x` to the `T` data type. The difference from [cast(x, T)](#type_conversion_function-cast) is that `accurateCast`
does not allow overflow of numeric types during cast if type value `x` does not fit bounds of type `T`. If the casted value is not representable in the target type, `accurateCast` throws an exception.
```
```suggestion
Converts `x` to the `T` data type. The differente from [cast(x, T)](#type_conversion_function-cast) is that `accurateCast`
does not allow overflow of numeric types during cast if type value `x` does not fit bounds of type `T`.
```

**Example**

Query:

``` sql
SELECT cast(-1, 'UInt8') as uint8; 
```

Result:

``` text
┌─uint8─┐
│   255 │
└───────┘
```

Query:

```sql
SELECT accurateCast(-1, 'UInt8') as uint8;
```

Result:

``` text
Code: 70. DB::Exception: Received from localhost:9000. DB::Exception: Value in column Int8 cannot be safely converted into type UInt8: While processing accurateCast(-1, 'UInt8') AS uint8.
```

## accurateCastOrNull(x, T) {#type_conversion_function-accurate-cast_or_null}

Converts input value `x` to the specified data type `T`. Always returns [Nullable](../../sql-reference/data-types/nullable.md) type and returns [NULL](../../sql-reference/syntax.md#null-literal) if the casted value is not representable in the target type.

**Syntax**

```sql
accurateCastOrNull(x, T)
```

**Parameters**

-   `x` — Input value.
-   `T` — The name of the returned data type.

**Example**

Query:

``` sql
SELECT toTypeName(accurateCastOrNull(5, 'UInt8'));
```

Result:

``` text
┌─toTypeName(accurateCastOrNull(5, 'UInt8'))─┐
│ Nullable(UInt8)                            │
└────────────────────────────────────────────┘
```

Query:

``` sql
SELECT

>     cast(-1, 'UInt8') as uint8,
>     cast(128, 'Int8') as int8,
>     cast('Test', 'FixedString(2)') as fixed_string;

Reviewer 2: ```suggestion
    accurateCastOrNull(-1, 'UInt8') as uint8,
    accurateCastOrNull(128, 'Int8') as int8,
    accurateCastOrNull('Test', 'FixedString(2)') as fixed_string;
```

> ``` sql
> SELECT
>     cast(-1, 'UInt8') as uint8,
>     cast(128, 'Int8') as int8,
>     cast('Test', 'FixedString(2)') as fixed_string;
> ```

Reviewer 2: ```suggestion
``` sql
SELECT
    accurateCastOrNull(-1, 'UInt8') as uint8,
    accurateCastOrNull(128, 'Int8') as int8,
    accurateCastOrNull('Test', 'FixedString(2)') as fixed_string;
```
```

Converts `x` to the `T` data type. 

The difference from [cast(x, T)](#type_conversion_function-cast) is that `accurateCast` does not allow overflow of numeric types during cast if type value `x` does not fit the bounds of type `T`. For example, `accurateCast(-1, 'UInt8')` throws an exception.

**Example**

Query:

``` sql
SELECT cast(-1, 'UInt8') as uint8; 
```

Result:

``` text
┌─uint8─┐
│   255 │
└───────┘
```

Query:

```sql
SELECT accurateCast(-1, 'UInt8') as uint8;
```

Result:

``` text
Code: 70. DB::Exception: Received from localhost:9000. DB::Exception: Value in column Int8 cannot be safely converted into type UInt8: While processing accurateCast(-1, 'UInt8') AS uint8.
```

## accurateCastOrNull(x, T) {#type_conversion_function-accurate-cast_or_null}

Converts input value `x` to the specified data type `T`. Always returns [Nullable](../../sql-reference/data-types/nullable.md) type and returns [NULL](../../sql-reference/syntax.md#null-literal) if the casted value is not representable in the target type.

**Syntax**

```sql
accurateCastOrNull(x, T)
```

**Parameters**

-   `x` — Input value.
-   `T` — The name of the returned data type.

**Returned value**

> -   The value in specified data type `T`.

Reviewer 2: ```suggestion
-   The value, converted to the specified data type `T`.
```
